internal/converter/internal/otelcolconvert/otelcolconvert.go

```diff
@@ -167,6 +168,39 @@ func appendConfig(file *builder.File, cfg *otelcol.Config) diag.Diagnostics {
 	// twice?
 	converterTable := buildConverterTable()
 
+	extensionTable := make(map[component.ID]componentID, len(cfg.Service.Extensions))
+
+	for _, ext := range cfg.Service.Extensions {
```

Comment:
We build the list of extensions 'activated' (defined in the service) as Flow components and keep a mapping of their OTel IDs to the blocks we've built.

There's no concept of multiple extensions per group or telemetry signals. Should we explain what we're doing here and why a little better?

Reply:
$0.02: I think needing to explain this in a GitHub comment is usually an indicator that it should be a code comment :) The next person reading this code isn't going to be able to easily find the PR and this comment explaining why this is happening here.

Reply:
Wrote this for myself but didn't follow up on it, thanks! 👀